moveit_simple/include/moveit_simple/moveit_simple.h

```diff
@@ -465,19 +468,71 @@ class OnlineRobot : public Robot
 
 
   /**
+   * @brief execute a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param goal - Joint trajectory goal which is a known 'Plan'
+   * @param collision_check - bool to turn check for collision on\off
+   * @throws <moveit_simple::ExecutionFailureException> (Execution failure)
+   * @throws <moveit_simple::IKFailException> (Conversion to joint trajectory failed)
+   * @throws <std::invalid_argument> (Trajectory "traj_name" not found)
+   * @throws <moveit_simple::CollisionDetected> (One of interpolated point is
+   * in Collision with itself or environment)
+   */
+  void executeKnownPlan(const std::string traj_name, 
+                   control_msgs::FollowJointTrajectoryGoal & goal,
+                   bool collision_check = false);
+
+
+  /**
+   * @brief plan out a given trajectory
+   * @param traj_name - name of trajectory to be executed (must be filled with
+   * prior calls to "addTrajPoint".
+   * @param goal - Joint trajectory goal
+   * @param collision_check - bool to turn check for collision on\off
+   * @throws <moveit_simple::ExecutionFailureException> (Execution failure)
+   * @throws <moveit_simple::IKFailException> (Conversion to joint trajectory failed)
```

Comment:
Can this still happen if we pass in a joint trajectory?

Reply:
No it cannot. But Plan returns a jointTrajectory.. the execute(...) function overload takes a jointTrajectory input and executes said traj. And yes, it will NOT throw an IKFailException. Please check out the documentation for that execute(...) function for more details on the exceptions it throws.